### PR TITLE
Fix for future effective bootstrap business address and directors also #94

### PR DESCRIPTION
**Pull request by BrandonSharratt**
*Issue:*https://github.com/bcgov/entity/issues/24689
Also https://github.com/bcgov/business-dashboard-ui/pull/94

*Description of changes:*
Accidently chucked 94 so readded (see link above for more info)
Also set forth pulling information on future effective incorporations to show address and directors on bootstrap filings if the information is available.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-dashboard-dev--pr-96-ugw271f0.web.app

File: src/pages/dashboard.vue [hunk 1]
```diff
@@ -357,7 +358,17 @@ const coaEffectiveDate = computed(() => {
...
+            v-if="bootstrapFiling?.filing?.incorporationApplication?.parties"
+            name="director"
+            :role-type="RoleTypeE.DIRECTOR"
+            :show-address="false"
+            :show-email="false"
+            :expand-top-item="true"
+          />
+          <p v-else>
+            {{ $t('text.filing.completeYourFiling') }}
+          </p>
```
Comment (severinbeauvais) [inline src/pages/dashboard.vue:371]: Can you please fix the alignment / whitespace for the current directors?

image

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-dashboard-dev--pr-96-ugw271f0.web.app

File: src/pages/dashboard.vue (hunk 1 above)
Comment (BrandonSharratt) [reply src/pages/dashboard.vue:371]: I've moved it to the left now and gotten rid of a lot of the extra bottom space

File: src/components/bcros/PartyInfo.vue [hunk 2]
```diff
@@ -2,30 +2,41 @@
...
   name: { type: String, required: true },
   roleType: { type: String, required: true },
   showEmail: { type: Boolean, required: true },
-  expandTopItem: { type: Boolean, default: false }
+  expandTopItem: { type: Boolean, default: false },
+  showAddress: { type: Boolean, default: true }
+})
+
+const disableExpand = computed(() => {
+  return !currentParties?.value?.parties && !!bootstrapFiling?.value?.filing?.incorporationApplication?.parties
```
Comment (hfekete) [inline src/components/bcros/PartyInfo.vue:26]: did you want to use && or maybe || when should expand be disabled ?
if no currentparties but incorporationApplication has parties set ?

File: src/components/bcros/PartyInfo.vue [hunk 3]
```diff
@@ -2,30 +2,41 @@
...
 })
 
 const partyItems = computed(() => {
   const items: BcrosAccordionItem[] = []
-
-  if (currentParties.value.parties) {
-    currentParties.value.parties.forEach((party) => {
+  const parties = currentParties?.value?.parties || bootstrapFiling?.value?.filing?.incorporationApplication?.parties
+  const disabled = !currentParties?.value?.parties &&
```
Comment (hfekete) [inline src/components/bcros/PartyInfo.vue:32]: NB: could reuse the above computed `disableExpand`

**Review by hfekete — APPROVED** [review]
If the `if condition` formula I mentioned in comment is good, looks good to merge from my point of view

File: src/components/bcros/PartyInfo.vue (hunk 2 above)
Comment (BrandonSharratt) [reply src/components/bcros/PartyInfo.vue:26]: I think we want && it makes it default to current functionality

**Pull request merged**